Arrow/Sources/Arrow/ArrowType.swift

```diff
@@ -165,14 +165,27 @@ public class ArrowTypeTimestamp: ArrowType {
     }
 }
 
-public class ArrowNestedType: ArrowType {
+public class ArrowTypeStruct: ArrowType {
     let fields: [ArrowField]
     public init(_ info: ArrowType.Info, fields: [ArrowField]) {
         self.fields = fields
         super.init(info)
     }
 }
 
+public class ArrowTypeList: ArrowType {
+    public let elementField: ArrowField
+
+    public init(_ elementField: ArrowField) {
+        self.elementField = elementField
+        super.init(ArrowType.ArrowList)
+    }
+
+    public convenience init(_ elementType: ArrowType, nullable: Bool = true) {
```

Comment:
it would probably be better to consistently use `isNullable` rather than nullable.